pipeline_dp/pipeline_operations.py

```diff
@@ -5,6 +5,7 @@
 import abc
 import apache_beam as beam
 import apache_beam.transforms.combiners as combiners
+from collections import defaultdict
```

Comment:
I recalled that by Google style guide, packages should be imported not classes. 
https://google.github.io/styleguide/pyguide.html#224-decision

Please update to 
import collections